celery/worker/consumer/consumer.py

```diff
@@ -323,12 +324,21 @@ def start(self):
             try:
                 blueprint.start(self)
             except self.connection_errors as exc:
-                # If we're not retrying connections, no need to catch
-                # connection errors
-                if not self.app.conf.broker_connection_retry:
-                    raise
+                # If we're not retrying connections, we need to properly shutdown or terminate
+                # the Celery main process instead of abruptly aborting the process without any cleanup.
+                is_connection_loss_on_startup = self.restart_count == 0
```

Comment:
`is_connection_loss_on_startup` sounds like it's a boolean.  Please initialize with `False` instead of 0.

Reply:
`is_connection_loss_on_startup` is being initialized with a boolean. Read 329 again.